### ISSUE TITLE
Guru Meditation Error: Core  1 panic'ed

### ISSUE THREAD

**Issue by darkmoon3d** [sparkfun/SparkFun_MS5803-14BA_Breakout_Arduino_Library#6]
Latest code update (1.1.3) is giving errors. Previous version seems to be running fine.

```
rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1184
load:0x40078000,len:13260
load:0x40080400,len:3028
entry 0x400805e4
Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x400d1be7  PS      : 0x00060830  A0      : 0x800d1f87  A1      : 0x3ffc52a0  
A2      : 0x00000000  A3      : 0x00000077  A4      : 0x00060a20  A5      : 0x00000001  
A6      : 0x007b7c80  A7      : 0x003fffff  A8      : 0x8008a73f  A9      : 0x3ffc5240  
A10     : 0x3ffc22c4  A11     : 0x3ffc22c4  A12     : 0x00000004  A13     : 0x00060a23  
A14     : 0x00060a23  A15     : 0x00000001  SAR     : 0x00000003  EXCCAUSE: 0x0000001c  
EXCVADDR: 0x00000040  LBEG    : 0x40086288  LEND    : 0x40086293  LCOUNT  : 0xffffffff  


Backtrace: 0x400d1be4:0x3ffc52a0 0x400d1f84:0x3ffc52d0 0x400d1fd8:0x3ffc52f0 0x400d2049:0x3ffc5310 0x400d161c:0x3ffc5330 0x400d409a:0x3ffc5360



ELF file SHA256: 4e22fcc77eb42575

Rebooting...
ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1184
load:0x40078000,len:13260
load:0x40080400,len:3028
entry 0x400805e4
Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.
```

**Comment by darkmoon3d** [comment]
> Latest code update (1.1.3) is giving errors. Previous version seems to be running fine.
> 
> ```
> rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
> configsip: 0, SPIWP:0xee
> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
> mode:DIO, clock div:1
> load:0x3fff0030,len:1184
> load:0x40078000,len:13260
> load:0x40080400,len:3028
> entry 0x400805e4
> Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.
> 
> Core  1 register dump:
> PC      : 0x400d1be7  PS      : 0x00060830  A0      : 0x800d1f87  A1      : 0x3ffc52a0  
> A2      : 0x00000000  A3      : 0x00000077  A4      : 0x00060a20  A5      : 0x00000001  
> A6      : 0x007b7c80  A7      : 0x003fffff  A8      : 0x8008a73f  A9      : 0x3ffc5240  
> A10     : 0x3ffc22c4  A11     : 0x3ffc22c4  A12     : 0x00000004  A13     : 0x00060a23  
> A14     : 0x00060a23  A15     : 0x00000001  SAR     : 0x00000003  EXCCAUSE: 0x0000001c  
> EXCVADDR: 0x00000040  LBEG    : 0x40086288  LEND    : 0x40086293  LCOUNT  : 0xffffffff  
> 
> 
> Backtrace: 0x400d1be4:0x3ffc52a0 0x400d1f84:0x3ffc52d0 0x400d1fd8:0x3ffc52f0 0x400d2049:0x3ffc5310 0x400d161c:0x3ffc5330 0x400d409a:0x3ffc5360
> 
> 
> 
> ELF file SHA256: 4e22fcc77eb42575
> 
> Rebooting...
> ets Jul 29 2019 12:21:46
> 
> rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
> configsip: 0, SPIWP:0xee
> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
> mode:DIO, clock div:1
> load:0x3fff0030,len:1184
> load:0x40078000,len:13260
> load:0x40080400,len:3028
> entry 0x400805e4
> Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.
> ```

**Comment by darkmoon3d** [comment]
Test code I'm using:

```
#include <Wire.h>
#include <SparkFun_MS5803_I2C.h>

MS5803 sensor(ADDRESS_LOW);

float temperature_c, temperature_f, pressure_abs;

void setup() {
  Wire.begin();
  
  Serial.begin(115200);

  sensor.reset();
  sensor.begin();

}

void loop() {

    // Check pressure value
  pressure_abs = sensor.getPressure(ADC_4096);
  if (pressure_abs <= 0.0 || pressure_abs > 14000.0) {
    // Pressure is outside valid range, trigger reset
    sensor.reset();
    return; // Exit loop
  }

  // Check temperature value
    temperature_c = sensor.getTemperature(CELSIUS, ADC_512);
  if (temperature_c < -40.0 || temperature_c > 85.0) {
    // Temperature is outside valid range, trigger reset
   sensor.reset();
    return; // Exit loop
  }

  // Read temperature from the sensor in deg C.
  temperature_c = sensor.getTemperature(CELSIUS, ADC_512);

  // Read temperature from the sensor in deg F.
  temperature_f = sensor.getTemperature(FAHRENHEIT, ADC_512);

  // Read pressure from the sensor in mbar.
  pressure_abs = sensor.getPressure(ADC_4096);

    Serial.print("Pressure: ");
    Serial.print(pressure_abs, 2); // Display pressure with 2 decimal places
    Serial.println(" mbar");

    Serial.print("Temperature: ");
    Serial.print(temperature_c, 2); // Display temperature with 2 decimal places
    Serial.print(" °C, ");
    Serial.print(temperature_f, 2); // Display temperature with 2 decimal places
    Serial.println(" °F");

  Serial.println(" "); // Padding between outputs

  delay(2000);
}

```

**Comment by PaulZC** [comment]
Hi,

In setup, please move the sensor.reset to after the sensor.begin. Does the code then work correctly?

Thanks,
Paul

**Comment by darkmoon3d** [comment]
That seemed to work. No problems now.

**Comment by PaulZC** [comment]
Ok - thanks!

I will release a proper fix for this tomorrow. Apologies for the inconvenience...

Best wishes,
Paul

**Comment by PaulZC** [comment]
Hi @darkmoon3d ,

This is resolved in [v1.1.4](https://github.com/sparkfun/SparkFun_MS5803-14BA_Breakout_Arduino_Library/releases/tag/v1.1.4). ```begin``` now calls ```reset``` internally. ```reset``` will return without crashing if called before ```begin```.

Apologies for the inconvenience,
Paul
